cmd/influx/write.go

```diff
@@ -240,6 +243,13 @@ func (writeFlags *writeFlagsType) createLineReader(ctx context.Context, cmd *cob
 		csvReader.RowSkipped = rowSkippedListener
 		r = csvReader
 	}
+	// throttle reader if requested
+	if writeFlags.LimitRate > 0.0 {
+		throttledReader := shapeio.NewReaderWithContext(r, ctx)
```

Comment:
The rate limit is enforced on the reading side, the [reader implementation](https://github.com/fujiwara/shapeio/blob/9c23d54b5329b479345e124153b904c0c1dcbded/shapeio.go#L72) reads `n` bytes and then it waits until these bytes are allowed to read in compliance with the rate limit. It relies upon the token bucket implementation, and it is IMHO proper. Your example works as expected to me. The `bust` property of the rate limiter does not influence the result. It could only if the number of bytes that are read at once would be more than a [billion](https://github.com/fujiwara/shapeio/blob/9c23d54b5329b479345e124153b904c0c1dcbded/shapeio.go#L11), the size of internal reading buffers is much lower (4096).

Reply:
The current implementation converts MB/min to bytes/sec and then ensures that the rate limit is never exceeded in every second. Assuming a limit of 5MB/5min, it does not allow to send 5MB quicker than in 5 minutes, it allows ~17.5kB every second. I don't consider that this is essentially wrong if the ultimate goal is to avoid "429 - limit exceeded". The tool can be now used repetitively in scripts to safely import more files. A refill period of 5 minutes (instead of 1 second) could be implemented, but that would certainly cause problems if the tool is executed repetitively. I could possibly implement a technique that would allow specifying the rate using both max size (5MB) and unit (5min), but that would require much more time to implement and test. 

Even better alternative (to specifying the rate) would be to react upon HTTP 429 errors during writing to InfluxDB and retry sending the message after the time specified in the server response (After HTTP header), a retry strategy would have to be implemented influx write.

The implementation provided herein is both quite simple for users and quite simple to implement and test, even though it does not provide the best throughput it possibly could. It seems better than '--pps' that is provided in influx v1. Maybe that I am lost in translation, can you please express your expectations for this PR to get approved?

Regarding the example that you provided, it is true, but the situation is more complex in this context. The basic unit that can be sent to the server is a protocol line. If you provide a CSV input (including a flux result CSV), the reader will not emit more than a single protocol line. More than a single line could be provided when reading protocol lines on input, so rather I improved the implementation with [LineReader](https://github.com/influxdata/influxdb/pull/19660/files#diff-a1dc84d7004d75c4e674f016b31ff0ceR252) to avoid bigger pauses than a fluent throttling really requires. The writer part (write/Batcher) then optimizes the network traffic by sending chunks of lines as soon as they extend 500kB in total. It IMHO does not matter much if that 500kB are delivered in smaller or bigger pieces, the rate will be +- the same when measured on input as a whole.